### DISCUSSION TITLE
How to renew the expired gpg keys after buying a new laptop?

### DISCUSSION THREAD

**Discussion by kanition** [community/community#22465] [New to GitHub]
<p>Hi~ I commit my code with signed gpg keys which have limited valid time. Now they are expired.<br>
What’s more, I bought a NEW laptop and have scrapped the old one!</p>
<p>So when I view my commit logs on GitHub, all are labeled “The key that signed this is expired”.<br>
Can I renew them with my new device? How to make it? Thank you for your advice :smile:</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>Key question here: Do you still have the private key, or did that get scrapped along with your old laptop? If you don’t have it, there’s nothing you can do.</p>
<p>If you have the key, you should be able to extend its lifetime. Then export the public key and (re-) add it to your account.</p>
<p>That said, it is expected that PGP keys expire at some point. Generally whenever that time approaches (a few weeks to months <em>before</em> it happens, preferably) you should decide: Is the key size and algorithm still sound? If yes, you can re-sign your key with an extended lifetime. If not (or you want a new key anyway), make a new key, sign it with the old one, and communicate it to anyone who might want to check your signatures (so they can validate it before the old key expires).</p>

**Comment by kanition** [comment]
<p>Thank you very much! Your solution is useful. :smile:<br>
I can’t have my old laptop along with me. So I have to accept the truth and tell others a new key will be made.</p>
